spring-kafka/src/main/java/org/springframework/kafka/annotation/RetryableTopicAnnotationProcessor.java

```diff
@@ -349,7 +363,8 @@ else if (resolved != null) {
 	private List<Class<? extends Throwable>> resolveClasses(Class<? extends Throwable>[] fromAnnot, String[] names,
 			String type) {
 
-		List<Class<? extends Throwable>> classes = new ArrayList<>(Arrays.asList(fromAnnot));
+		List<Class<? extends Throwable>> classes = new ArrayList<>(fromAnnot.length + names.length);
+		Collections.addAll(classes, fromAnnot);
```

Comment:
the above changes tries to improve on:

- code readability; 
- a little bit perf opt (super minor; JMH shows it boosts perf by 100%)

Again, let me know whether we need to this in the first place